include/mxnet/lib_api.h

```diff
@@ -1102,7 +1232,12 @@ extern "C" {
                   const int64_t** outshapes, int* outdims, void** outdata, int* outtypes,
                   size_t* outIDs, const char** outdev_type, int* outdev_id, int num_out,
                   xpu_malloc_t cpu_malloc, void* cpu_alloc,
-                  xpu_malloc_t gpu_malloc, void* gpu_alloc, void* cuda_stream) {
+                  xpu_malloc_t gpu_malloc, void* gpu_alloc, void* cuda_stream,
+                  sparse_malloc_t sparse_malloc, void* sparse_alloc,
+                  int* instypes, int* outstypes, void** in_indices, void** out_indices,
+                  void** in_indptr, void** out_indptr,
+                  int64_t* in_indices_shapes, int64_t* out_indices_shapes, 
+                  int64_t* in_indptr_shapes, int64_t* out_indptr_shapes) {
```

Comment:
There are too many arguments in a function. We can use C structure to reduce the number of arguments later in other PR.

Reply:
added to https://github.com/apache/incubator-mxnet/issues/17006

Reply:
As Sam mentioned, we could add it as a to-do task for next step.